src/pykeen/models/unimodal/ermlp.py

```diff
@@ -2,24 +2,94 @@
 
 """Implementation of ERMLP."""
 
+import math
 from typing import Optional
 
 import torch
-import torch.autograd
 from torch import nn
 
-from ..base import EntityRelationEmbeddingModel
+from ..base import InteractionFunction, SimpleVectorEntityRelationEmbeddingModel
 from ...losses import Loss
 from ...regularizers import Regularizer
 from ...triples import TriplesFactory
 from ...typing import DeviceHint
 
 __all__ = [
     'ERMLP',
+    'ERMLPInteractionFunction',
 ]
 
 
-class ERMLP(EntityRelationEmbeddingModel):
+class ERMLPInteractionFunction(InteractionFunction):
+    """
+    Interaction function of ER-MLP.
+
+    .. math ::
+        f(h, r, t) = W_2 ReLU(W_1 cat(h, r, t) + b_1) + b_2
+    """
+
+    def __init__(
+        self,
+        embedding_dim: int,
+        hidden_dim: int,
+    ):
+        """
+        Initialize the interaction function.
+
+        :param embedding_dim:
+            The embedding vector dimension.
+        :param hidden_dim:
+            The hidden dimension of the MLP.
+        """
+        super().__init__()
+        """The multi-layer perceptron consisting of an input layer with 3 * self.embedding_dim neurons, a  hidden layer
+           with self.embedding_dim neurons and output layer with one neuron.
+           The input is represented by the concatenation embeddings of the heads, relations and tail embeddings.
+        """
+        self.head_to_hidden = nn.Linear(in_features=embedding_dim, out_features=hidden_dim, bias=False)
+        self.rel_to_hidden = nn.Linear(in_features=embedding_dim, out_features=hidden_dim, bias=True)
+        self.tail_to_hidden = nn.Linear(in_features=embedding_dim, out_features=hidden_dim, bias=False)
+        self.activation = nn.ReLU()
+        self.hidden_to_score = nn.Linear(in_features=hidden_dim, out_features=1, bias=True)
+
+    def forward(
+        self,
+        h: torch.FloatTensor,
+        r: torch.FloatTensor,
+        t: torch.FloatTensor,
+        **kwargs,
```

Comment:
can you just omit **kwargs here?

Reply:
At least my IDE then complains about not implementing the interface :sweat_smile:

Reply:
Maybe we can just solve this by an annotation `no_more_kwargs` which wraps the function with the generic **kwargs intereface, and does the check for additional kwargs.

Reply:
yuck, maybe we should just make the base class h,r,t,*args,**kwargs so the subclasses can be anything